packages/core/src/browser/keybinding.ts

```diff
@@ -279,23 +339,43 @@ export class KeybindingRegistry {
         return result;
     }
 
+    /**
+     * Returns wether a binding is being shadowed or not inside scope
+     *
+     * @param binding binding to look if shadowed
+     * @param scope scope where to look
+     */
+    protected isKeybindingShadowed(bindings: Keybinding[], binding: Keybinding): boolean {
+        for (const registeredBinding of bindings) {
+            try {
+                const registeredKeySequence = KeySequence.parse(registeredBinding.keybinding);
+                const keySequence = KeySequence.parse(binding.keybinding);
+                const comparaison = KeySequence.compare(keySequence, registeredKeySequence);
+
+                if (comparaison === KeySequence.CompareResult.SHADOW) {
+                    return true;
+                }
+            } catch (error) {
+                this.logger.warn(error);
+                continue;
+            }
+        }
+        return false;
+    }
+
     /**
      * Returns true if a keybinding is shadowed in a more specific scope i.e bound in user scope but remapped in
      * workspace scope means the user keybinding is shadowed.
-     * @param scope scope of the current keybinding
+     * @param startScope scope of the current keybinding
      * @param binding keybinding that will be checked in more specific scopes
      */
-    isKeybindingShadowed(scope: KeybindingScope, binding: Keybinding): boolean {
-        if (scope >= KeybindingScope.END) {
-            return false;
-        }
-
-        const nextScope = ++scope;
-
-        if (this.getScopedKeybindingsForCommand(nextScope, binding.command).length > 0) {
-            return true;
+    isKeybindingShadowedByAny(binding: Keybinding, startScope: KeybindingScope = KeybindingScope.DEFAULT): boolean {
```

Comment:
I really like the doc above that says what this function does, but I feel like the name is misleading as I would expect it to check if it's shadowed by any starting from the most specific scope. Either make it more generic but expect the user to check the option, or very specific or more clear. Maybe rename it to `isKeybindingShadowed`... Wait that's used just above... Anyway is this function even used anywhere? If not I would suggest removing it.